crates/fluvio-extension-common/src/common/time.rs

```diff
@@ -0,0 +1,141 @@
+use chrono::{DateTime, Utc};
+
+const SIMPLE_DATE_FORMAT: &str = "%Y/%m/%d";
+const EXACT_FORMAT: &str = "%Y-%m-%d %H:%M:%S %Z";
+
+pub struct TimeElapsedFormatter {
+    bot: DateTime<Utc>,
+}
+
+impl Default for TimeElapsedFormatter {
+    fn default() -> Self {
+        Self { bot: Utc::now() }
+    }
+}
+
+impl TimeElapsedFormatter {
+    pub fn new(bot: DateTime<Utc>) -> Self {
+        Self { bot }
+    }
+
+    pub fn time_since_str(&self, date: DateTime<Utc>, exact: bool) -> String {
+        let duration = self.bot.signed_duration_since(date);
+
+        if exact {
+            return date.format(EXACT_FORMAT).to_string();
+        }
+
+        if duration.num_weeks() >= 1 {
+            return date.format(SIMPLE_DATE_FORMAT).to_string();
+        }
+
+        if duration.num_days() >= 2 {
+            return format!("{} days ago", duration.num_days());
+        }
+
+        if duration.num_days() == 1 {
+            return String::from("Yesterday");
+        }
+
+        if duration.num_hours() >= 1 && duration.num_hours() < 24 {
+            return format!("{} hours ago", duration.num_hours());
+        }
+
+        if duration.num_minutes() >= 1 && duration.num_minutes() < 59 {
+            return format!("{} minutes ago", duration.num_minutes());
+        }
+
+        if duration.num_minutes() < 1 {
+            return String::from("Just now");
+        }
+
+        date.format(SIMPLE_DATE_FORMAT).to_string()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use chrono::{DateTime, Days, TimeZone, Timelike, Utc};
+
+    use crate::time::{TimeElapsedFormatter, EXACT_FORMAT, SIMPLE_DATE_FORMAT};
+
+    fn december_17_1989_10_30_00() -> DateTime<Utc> {
+        Utc.with_ymd_and_hms(1989, 12, 17, 10, 30, 0)
+            .unwrap()
+            .to_utc()
+    }
+
+    fn make_fixed_time_elapsed_fmtr() -> TimeElapsedFormatter {
+        TimeElapsedFormatter::new(december_17_1989_10_30_00())
+    }
+
+    #[test]
+    fn determines_exact_date() {
+        let next_min = december_17_1989_10_30_00().minute() - 30;
+        let date = december_17_1989_10_30_00()
+            .with_minute(next_min)
+            .expect("failed to set min");
+        let formatted = make_fixed_time_elapsed_fmtr().time_since_str(date, true);
+
+        assert_eq!(formatted, date.format(EXACT_FORMAT).to_string());
```

Comment:
test should have assertion like this: 
````
assert_eq!(date,"December 1989");
````

Reply:
Of course! Here is the update: [0a4d54a](https://github.com/infinyon/fluvio/pull/4203/commits/0a4d54ac3df22a05cc769e0504f477f0d3e3a2a3)